spec/index.adoc

```diff
@@ -1738,6 +1738,113 @@ Any user can top-up any canister this way.
+than the Bitcoin component, which implies that it is advisable to set the number of confirmations to a reasonably large value, such as 6,
+to gain confidence in the correctness of the returned UTXOs.
+
+There is an upper bound of 100,000 UTXOs that can be returned in a single request. For addresses that contain sufficiently large UTXOs,
+the partial set of the address' UTXOs are returned along with a page reference.
+
+In the second case, a page reference (a series of bytes) must be provided, which instructs the Bitcoin functionality in the replica to
+collect UTXOs starting from the corresponding "page".
+
+A `get_utxos_request` without the optional `filter` results in a request that
+considers the full blockchain, which is equivalent to setting
+`min_confirmations` to 0.
+
+The recommended workflow is to issue a request with the desired number of confirmations.
+If the `next_page` field in the response is not empty, there are more UTXOs than in the returned vector.
+In that case, the `page` field should be set to the `next_page` bytes in the subsequent request to obtain the next batch of UTXOs.
+
+[#ic-bitcoin_get_balance]
+=== IC method `bitcoin_get_balance`
+
+Given a `get_balance_request`, which must specify a Bitcoin address and a Bitcoin network (`mainnet` or `testnet`), the function returns the current balance of this address in `Satoshi` (10^8 Satoshi = 1 Bitcoin) in the specified Bitcoin network. The same address formats as for link:#ic-bitcoin_get_utxos[`bitcoin_get_utxos`] are supported.
+
+If the address is malformed, the call is rejected.
+
+The optional `min_confirmations` parameter can be used to limit the set of considered UTXOs
+for the calculation of the balance to those with at least the provided number of confirmations
+in the same manner as for the link:#ic-bitcoin_get_utxos[`bitcoin_get_utxos`] call.
+
+Given an address and the optional `min_confirmations` parameter, `bitcoin_get_balance` iterates over
+all UTXOs, i.e., the same balance is returned as when calling
+link:#ic-bitcoin_get_utxos[`bitcoin_get_utxos`]
+for the same address and the same number of confirmations and, if necessary,
+using pagination to get all UTXOs for the same tip hash.
+
+[#ic-bitcoin_send_transaction]
+=== IC method `bitcoin_send_transaction`
+
+Given a `send_transaction_request`, which must specify a `blob` of a Bitcoin transaction and a Bitcoin network (`mainnet` or `testnet`), several checks are performed:
+
+- The transaction is well formed.
+- The transaction only consumes unspent outputs with respect to the current (longest)
+blockchain, i.e., there is no block on the (longest) chain that consumes any of these outputs.
+- There is a positive transaction fee.
+
+If at least one of these checks fails, the call is rejected.
+
+If the transaction passes these tests, the transaction is forwarded to the
+specified Bitcoin network.
+
+The Bitcoin component of the replica periodically forwards the transaction
```

Comment:
The spec so far does not mention any implementation specific components (including the `replica`). Sometimes the spec makes reference to the "most likely behavior", like in the case of heartbeats. 

I would strongly recommend that the behavior is specified based on the abstract state of the Internet Computer only. Does that make sense?

Reply:
I agree that we should not be mentioning the replica here, as its only one implementation of the spec. WDYT @THLO?

Reply:
I agree.
Does one of you have a suggestion on how to rephrase it?